### PR TITLE
iplemented CustomGlobalExceptionHandler class and added meaningful va…

### PR DESCRIPTION
**Pull request by DPotapov93**
…lidation

### CODE REVIEW

File: src/main/java/mate/academy/onlinebookstore01/dto/CreateBookRequestDto.java [hunk 1]
```diff
@@ -1,14 +1,24 @@
...
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
 import java.math.BigDecimal;
 import lombok.Data;
 
 @Data
 public class CreateBookRequestDto {
+    @Min(1)
     private Long id;
+    @NotNull
```
Comment (forzen123212) [inline src/main/java/mate/academy/onlinebookstore01/dto/CreateBookRequestDto.java:13]: Add min and max length

File: src/main/java/mate/academy/onlinebookstore01/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -1,14 +1,24 @@
...
     private String title;
+    @NotNull
```
Comment (forzen123212) [inline src/main/java/mate/academy/onlinebookstore01/dto/CreateBookRequestDto.java:15]: same

**Review by forzen123212 — APPROVED** [review]
Good job, just some minor changes are need to be added

File: src/main/java/mate/academy/onlinebookstore01/exception/CustomGlobalExceptionHandler.java
Comment (d1sam) [inline src/main/java/mate/academy/onlinebookstore01/exception/CustomGlobalExceptionHandler.java:1]: Please also handle your own `EntityNotFoundException.class`

File: src/main/java/mate/academy/onlinebookstore01/exception/CustomGlobalExceptionHandler.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            HttpStatusCode status,
+            WebRequest request
+    ) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.BAD_REQUEST);
+        List<String> errors = ex.getBindingResult().getAllErrors().stream()
+                .map(this::getErrorMessage)
+                .toList();
+        body.put("errors", errors);
```
Comment (d1sam) [inline src/main/java/mate/academy/onlinebookstore01/exception/CustomGlobalExceptionHandler.java:33]: String literals can be as constants

File: src/main/java/mate/academy/onlinebookstore01/dto/CreateBookRequestDto.java [hunk 4]
```diff
@@ -1,14 +1,24 @@
...
     private String author;
+    @NotNull
+    @Size(min = 10, max = 13)
     private String isbn;
+    @NotNull
+    @Min(0)
```
Comment (d1sam) [inline src/main/java/mate/academy/onlinebookstore01/dto/CreateBookRequestDto.java:21]:
```suggestion
    @Positive
```

File: src/main/java/mate/academy/onlinebookstore01/dto/CreateBookRequestDto.java [hunk 5]
```diff
@@ -1,14 +1,24 @@
...
 @Data
 public class CreateBookRequestDto {
+    @Min(1)
     private Long id;
+    @NotNull
     private String title;
+    @NotNull
     private String author;
+    @NotNull
+    @Size(min = 10, max = 13)
```
Comment (d1sam) [inline src/main/java/mate/academy/onlinebookstore01/dto/CreateBookRequestDto.java:18]: ISBN has fixed format, you can use [Hibernate validators](https://www.baeldung.com/hibernate-validator-constraints) or rewrite your regexp by [this](https://howtodoinjava.com/java/regex/java-regex-validate-international-standard-book-number-isbns/) guide

File: src/main/java/mate/academy/onlinebookstore01/dto/CreateBookRequestDto.java [hunk 6]
```diff
@@ -1,14 +1,24 @@
...
+import jakarta.validation.constraints.Size;
 import java.math.BigDecimal;
 import lombok.Data;
 
 @Data
 public class CreateBookRequestDto {
+    @Min(1)
     private Long id;
+    @NotNull
     private String title;
```
Comment (d1sam) [inline src/main/java/mate/academy/onlinebookstore01/dto/CreateBookRequestDto.java:14]: I would also validate a maximum size of text fields, to not exceed the DB column restrictions.

File: src/main/java/mate/academy/onlinebookstore01/dto/CreateBookRequestDto.java [hunk 7]
```diff
@@ -1,14 +1,24 @@
...
+    @NotNull
     private String author;
```
Comment (d1sam) [inline src/main/java/mate/academy/onlinebookstore01/dto/CreateBookRequestDto.java:16]: I would also validate a maximum size of text fields, to not exceed the DB column restrictions.

File: src/main/java/mate/academy/onlinebookstore01/dto/CreateBookRequestDto.java
Comment (d1sam) [inline src/main/java/mate/academy/onlinebookstore01/dto/CreateBookRequestDto.java:1]: Also provide custom messages to validation annotations

**Review by d1sam — CHANGES_REQUESTED** [review]
Looks good, but see a couple of comments.

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**